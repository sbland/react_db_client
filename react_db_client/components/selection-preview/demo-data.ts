import { filterTypes } from '@react_db_client/constants.client-types';

export const defaultProps = {
    headings: [
      {
        uid: 'uid',
        label: 'Uid',
        type: filterTypes.text,
      },
      {
        uid: 'label',
        label: 'Label',
        type: filterTypes.text,
      },
      {
        uid: 'other',
        label: 'Other',
        type: 'other',
      },
    ],
    currentSelectionData: {
      label: 'Foo',
      uid: 'foo',
      other: 'custom',
    },
    customParsers: {
      other: (value) => `{${value}}`,
    },
  };

