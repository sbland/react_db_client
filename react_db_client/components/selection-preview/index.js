export { SelectionPreview } from './selection-preview';
