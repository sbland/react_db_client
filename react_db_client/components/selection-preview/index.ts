export { SelectionPreview } from './selection-preview';
export type { ISelectionPreviewProps } from './selection-preview';
