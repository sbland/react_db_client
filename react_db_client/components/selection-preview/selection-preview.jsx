import React from 'react';
import PropTypes from 'prop-types';

import { stringifyData } from '@samnbuk/react_db_client.helpers.data-processing';

import './style.scss';

export const SelectionPreview = ({ headings, currentSelectionData, customParsers }) => {
  const cellData = currentSelectionData
    ? headings.map((heading) => {
        const cleanedValue = stringifyData(
          currentSelectionData[heading.uid],
          heading,
          customParsers
        );
        return [heading.uid, heading.label, cleanedValue];
      })
    : [];
  return (
    <div className="flexHoriz">
      <h3>Selection Preview</h3>
      <ul className="selectionPreview_listWrap flexGrow">
        {cellData.map(([uid, label, value]) => {
          return (
            <li className="selectionPreview_listItem" key={uid}>
              <label>{label}: </label>
              {value}
            </li>
          );
        })}
      </ul>
    </div>
  );
};

SelectionPreview.propTypes = {
  currentSelectionData: PropTypes.shape({
    uid: PropTypes.string,
    label: PropTypes.string,
  }).isRequired,
  headings: PropTypes.arrayOf(
    PropTypes.shape({
      uid: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  customParsers: PropTypes.objectOf(PropTypes.func),
};

SelectionPreview.defaultProps = {
  customParsers: {},
};
