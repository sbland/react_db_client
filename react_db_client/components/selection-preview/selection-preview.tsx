import React from 'react';
import PropTypes from 'prop-types';

import { stringifyData } from '@react_db_client/helpers.data-processing';
import { EFilterType, Uid } from '@react_db_client/constants.client-types';
import {
  SelectionPreviewLabelStyle,
  SelectionPreviewList,
  SelectionPreviewListItem,
} from './styles';

export interface IHeading {
  uid: Uid;
  label: string | React.ReactNode;
  type: EFilterType | string;
}

export type CustomParser = (value: any) => any;

export interface ISelectionPreviewProps {
  headings: IHeading[];
  currentSelectionData: { [id: string]: any };
  customParsers?: { [key: string]: CustomParser };
  listStyleOverride?: Partial<React.CSSProperties>;
}

export const SelectionPreview = ({
  headings,
  currentSelectionData,
  customParsers,
  listStyleOverride = {},
}: ISelectionPreviewProps) => {
  const cellData = currentSelectionData
    ? headings.map((heading) => {
        const cleanedValue = stringifyData(
          currentSelectionData[heading.uid],
          heading,
          customParsers
        );
        return [heading.uid, heading.label, cleanedValue];
      })
    : [];
  return (
    <div className="flexHoriz">
      <SelectionPreviewList style={{ ...listStyleOverride }}>
        {cellData.map(([uid, label, value]) => {
          return (
            <SelectionPreviewListItem key={uid}>
              <SelectionPreviewLabelStyle>{label}: </SelectionPreviewLabelStyle>
              {value}
            </SelectionPreviewListItem>
          );
        })}
      </SelectionPreviewList>
    </div>
  );
};

SelectionPreview.propTypes = {
  currentSelectionData: PropTypes.shape({
    uid: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    label: PropTypes.string,
  }).isRequired,
  headings: PropTypes.arrayOf(
    PropTypes.shape({
      uid: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  customParsers: PropTypes.objectOf(PropTypes.func),
};

SelectionPreview.defaultProps = {
  customParsers: {},
};
