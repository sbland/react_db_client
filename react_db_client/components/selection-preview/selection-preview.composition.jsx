import React from 'react';
import { SelectionPreview } from './selection-preview';

export const BasicSelectionPreview = () => (
  <SelectionPreview text="hello from SelectionPreview" />
);
