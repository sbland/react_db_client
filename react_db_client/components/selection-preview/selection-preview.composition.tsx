import React from 'react';
import { SelectionPreview } from './selection-preview';
import { defaultProps } from './demo-data';


export const BasicSelectionPreview = () => <SelectionPreview {...defaultProps} />;
