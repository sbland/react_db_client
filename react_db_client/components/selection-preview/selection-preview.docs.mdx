---
description: 'A SelectionPreview component.'
labels: ['ui']
---

import { SelectionPreview } from './selection-preview';

## React Component for rendering meta data of selection json

A basic div that renders some meta json data cleanly
