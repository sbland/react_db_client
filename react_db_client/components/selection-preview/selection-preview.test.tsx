import React from 'react';
import { render } from '@testing-library/react';

import * as compositions from './selection-preview.composition';

describe('Selection Preview', () => {
  describe('Compositions', () => {
    Object.entries(compositions).forEach(([name, Composition]) => {
      test(name, async () => {
        render(<Composition />);
        // @ts-ignore
        if (Composition.waitForReady) await Composition.waitForReady();
      });
    });
  });
  describe('Unit Tests', () => {
    test.todo('should show meta info on document');
  });
});
