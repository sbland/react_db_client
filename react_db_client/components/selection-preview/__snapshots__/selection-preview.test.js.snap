// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionPreview Matches Snapshot 1`] = `
<div
  className="flexHoriz"
>
  <h3>
    Selection Preview
  </h3>
  <ul
    className="selectionPreview_listWrap flexGrow"
  >
    <li
      className="selectionPreview_listItem"
      key="uid"
    >
      <label>
        Uid
        : 
      </label>
      foo
    </li>
    <li
      className="selectionPreview_listItem"
      key="label"
    >
      <label>
        Label
        : 
      </label>
      Foo
    </li>
    <li
      className="selectionPreview_listItem"
      key="other"
    >
      <label>
        Other
        : 
      </label>
      {custom}
    </li>
  </ul>
</div>
`;
